packages/babel-parser/src/plugins/flow.js

```diff
@@ -901,9 +924,42 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           );
         }
         this.expect(tt.ellipsis);
-        node.argument = this.flowParseType();
+        switch (this.state.type) {
+          case tt.comma:
+          case tt.semi:
+          case tt.braceR:
+          case tt.braceBarR:
+            this.flowObjectTypeSemicolon();
+            switch (this.state.type) {
+              case tt.braceR:
```

Comment:
Nit: I think that this code would be more readable without nested switch statemets:

```js
let isInexactToken = this.eat(tt.comma) || this.eat(tt.semi);

if (this.match(tt.braceR)) {
  if (allowInexact) return true;
  this.unexpected(null, "Explicit inexact syntax is only allowed inside inexact objects");
}

if (this.match(tt.braceBarR)) {
  this.unexpected(null, "Explicit inexact syntax cannot appear inside an explicit exact object type");
}

if (isInexactToken) {
  this.unexpected(null, "Explicit inexact syntax must appear at the end of an inexact object");
}

node.argument = this.flowParseType();
return this.finishNode(node, "ObjectTypeSpreadProperty");
```

Do what you prefer though

Reply:
I like this better, will follow up